lms/djangoapps/instructor/views/api.py

```diff
@@ -2708,7 +2704,7 @@ def start_certificate_regeneration(request, course_id):
         )
 
     # Check if the selected statuses are allowed
-    allowed_statuses = GeneratedCertificate.get_unique_statuses(course_key=course_key, flat=True)
+    allowed_statuses = [CertificateStatuses.downloadable, CertificateStatuses.error, CertificateStatuses.notpassing]
```

Comment:
@stvstnfrd all the available statuses can be seen at https://github.com/edx/edx-platform/blob/master/lms/djangoapps/certificates/models.py#L75-L88

It was dynamic during first implementation of Certificate Regenerate UI as suggested by @mattdrayer but after UI revision only three options were added for certificate regeneration ui

1. Regenrate for learners who have already received certificates (#)
2. Regenerate for leaners who have not received certificates (#)
3. Regenerate for learners in an error state (#)

As only these statuses are likely to be valid for certificate regeneration other statuses would either be not present or would not be valid for certificate regeneration.